### DISCUSSION TITLE
Colors and format of spot output

### DISCUSSION THREAD

**Discussion by knutov** [umputun/spot#149] [General]
## Colors 
There is no customization of spot output now. And I think it's nice to have it. Personally I'd like to have ability to change colors of the output and ability to have output in json/yaml formats (to parse it with some external tools).

The problem with colors is when I run task with multiple servers the output with current colors does not look human readable - as lines are mostly red and blue. It's hard to understand which color relates to which server:

![изображение](https://github.com/umputun/spot/assets/255869/6f8d3ea3-d775-4d42-8e43-532a54117cc4)

## Output format

With output - I'd like to get some yaml output via export and print to to the output to be read by human. For now every screen line will be prepended by server address, it takes space. Whould be nice to have not columnar ouput but something like one line with server address and may be some tech data and next line with `echo:` output. Is it easy to add?

**Comment by umputun** [comment]
The color of output for a given server is set [like this](https://github.com/umputun/spot/blob/cc10e4afa0b56fff2f2c560529403bfe82003256/pkg/executor/executor.go#L133-L133). As you can see each server get a persistent color from the table based on the `host` value and it includes all the colors here. Red and blue in your case are set because of the targets you are running spot for, it can be different colors for other hosts.

> the output with current colors does not look human readable

I don't really see unreadable output on your screenshot. The dark red and bright red (FgRed & FgHiRed) look different enough to me. One thing that probably make sense - adding a param/env-var to disable colorization 

> Whould be nice to have not columnar output

This is not easy to do if a task running concurrently and different targets are mixed in the output. In addition, this may double lines, and generally, I don't like the idea of such a "stateless" print where the user needs to check the previous line to understand the actual output

> With output - I'd like to get some yaml output via export and print to to the output to be read by human.

I don't understand what export and print for "some yaml output" means in this context. Is it about `--gen`? If so, you already can provide any custom template

**Reply by knutov** [reply]
output - after export some variable can contain yaml or json. I'd like to see it from new line, without prepening any text before.

Current output in spot consist of some message prepending left column like this:

```
[x1 x1.server.net:22] completed command "uptime_get" {script: /bin/sh -c /tmp/.spot-1237993603777309184/spot-script3205487850} (1.014s)
```

If it will be possible to have as

```yaml
  - server: x1 x1.server.net:22
    message: "completed command "uptime_get" {script: /bin/sh -c /tmp/.spot-1237993603777309184/spot-script3205487850} (1.014s)"
```

or even better:

```yaml
  - server: 
      name: x1.server.net:22
      short: x1
    output: 
      message: "completed command "uptime_get"
      script: /bin/sh -c /tmp/.spot-1237993603777309184/spot-script3205487850
      time: 1.014s
      script_output: ...
```

it will be possible to process it with some external processor and use both desired color generator and do not prepend lines with column with server address.
